pymatgen/transformations/advanced_transformations.py

```diff
@@ -40,15 +40,16 @@
 )
 from pymatgen.transformations.transformation_abc import AbstractTransformation
 
-if TYPE_CHECKING:
-    from collections.abc import Iterable, Sequence
-    from typing import Any
-
 try:
     import hiphive
 except ImportError:
     hiphive = None
 
+if TYPE_CHECKING:
+    from collections.abc import Iterable, Sequence
+    from typing import Any
```

Comment:
I hope I'm understanding this correctly,

```python
try:
    import hiphive
except ImportError:
    hiphive = None
```
Still is the "standard" import block.

While:
```python
if TYPE_CHECKING:
    from collections.abc import Iterable, Sequence
    from typing import Any
```
is the "conditional" import block specific to `TYPE_CHECKING`.

As such the order should be adjusted such that "standard" imports go together.